### ISSUE TITLE
List of relevant resources

### ISSUE THREAD

**Issue by barisozmen** [barisozmen/deepaugment#7]
### Journal papers
* AutoAugment: Learning Augmentation Policies from Data ([link](https://arxiv.org/pdf/1805.09501.pdf))
* Smart Augmentation - Learning an Optimal Data Augmentation Strategy ([link](https://arxiv.org/abs/1703.08383))
* Adaptive data augmentation for image classification ([link](https://infoscience.epfl.ch/record/218496/files/ICIP_CAMERAREADY_2715.pdf))
* Learning to Compose Domain-Specific Transformations for Data Augmentation ([link](https://hazyresearch.github.io/snorkel/blog/tanda.html))
    * Cited by AutoAugment paper

### AutoAugment implementations
* Official ([github](https://github.com/tensorflow/models/tree/master/research/autoaugment))
    * Tensorflow used 
    * does not have controller (RL) part
* Unofficial ([github](https://github.com/rpmcruz/autoaugment))
    * Keras used
    * Have the controller part
* Unofficial exploration on jupyter notebook ([github](https://github.com/DeepVoltaire/AutoAugment/blob/master/AutoAugment_Exploration.ipynb))
    * Very good for learning
    * Pytorch used
    * Doesn't implement the whole
* In tensorflow-hub ([link](https://tfhub.dev/s?q=autoaugment))
### Blogs
* [https://towardsdatascience.com/how-to-improve-your-image-classifier-with-googles-autoaugment-77643f0be0c9](https://towardsdatascience.com/how-to-improve-your-image-classifier-with-googles-autoaugment-77643f0be0c9)
* Comparing accuracy increase of various augmentation methods ([https://medium.com/nanonets/how-to-use-deep-learning-when-you-have-limited-data-part-2-data-augmentation-c26971dc8ced](https://medium.com/nanonets/how-to-use-deep-learning-when-you-have-limited-data-part-2-data-augmentation-c26971dc8ced))
    * NanoNets
* A bag of tricks for image classification ([https://towardsdatascience.com/a-big-of-tricks-for-image-classification-fec41eb28e01](https://towardsdatascience.com/a-big-of-tricks-for-image-classification-fec41eb28e01))
    * Occlusion (putting grey rectangle to some areas of image) as a data augmentation method.
* Deep neuro-evolution (DL + GA) ([https://eng.uber.com/deep-neuroevolution/](https://eng.uber.com/deep-neuroevolution/)) ([github](https://github.com/uber-research/deep-neuroevolution))
    * Ekin Dogus suggested at this [video (12:17)](https://www.youtube.com/watch?v=InyC8KQYfNM)
    * Uber AI team hybridizing deep learning with genetic algorithms
### Videos
* Seminar presentation ([Youtube](https://www.youtube.com/watch?v=qkl_7f4XO7A))
    * distills the situation and the model well!
    * discussion part gives a good critic, data augmentation might overfit for the validation set

### Libraries
* [Tanda](https://github.com/HazyResearch/tanda) Learning to Compose Domain-Specific Transformations for Data Augmentation
* [Automold](https://github.com/UjjwalSaxena/Automold--Road-Augmentation-Library) road augmentation library for self-driving cars
* [Albumentations](https://github.com/albu/albumentations) general data augmentation library
* [Autosat](https://github.com/mapbox/robosat) Semantic segmentation on aerial and satellite imagery. Extracts features such as: buildings, parking lots, roads, water, clouds
* [NAS](https://github.com/titu1994/neural-architecture-search/blob/master/controller.py)
* [autokeras](https://autokeras.com/)

### Ideas
* How to make AutoAugment scalable?
    * By training models with 50-100 epochs in the beginning with original data. And then training it with suggested augmentations!
        * Training times of various models and datasets ([https://dawn.cs.stanford.edu/benchmark/)](https://dawn.cs.stanford.edu/benchmark/)
        * The DAWNbench paper ([https://dawn.cs.stanford.edu/benchmark/papers/nips17-dawnbench.pdf)](https://dawn.cs.stanford.edu/benchmark/papers/nips17-dawnbench.pdf)
* Transfer learning (TL) for decreasing computation time
* Transfer learning and genetic algorithms
* A case for using Random Search instead of RL:
    * *"Typically random search algorithms sacrifice a guarantee of optimality for finding a good solution quickly with convergence results in probability." *([Random Search Algorithms](http://courses.washington.edu/inde510/516/AdapRandomSearch4.05.2009.pdf))
    * Also see: "Simple random search provides a competitive approach to reinforcement learning" ([https://arxiv.org/abs/1803.07055](https://arxiv.org/abs/1803.07055))
   *  Data augmentation guided by cooccurrences of objects
   * (https://www.researchgate.net/publication/328899557_Learning_data_augmentation_policies_using_augmented_random_search) paper claims that they get better results than AutoAugment by tweaking its parameters
